build.sbt

```diff
@@ -27,10 +27,6 @@ inThisBuild(
       // https://github.com/scala/bug/issues/10448
       "-Ywarn-unused:imports"
     ),
-    addCompilerPlugin(
```

Comment:
I add it per project, because it fails when added to a dotty project.

Reply:
The following setting should work
```scala
libraryDependencies ++= {
  if (isScala2) List(compilerPlugin(...))
  else List()
}
```

Reply:
Not under `inThisBuild` however, it needs to be added as a shared setting

```scala
val sharedSettings = List(
  libraryDependencies ++= {
    if (isScala2) List(compilerPlugin(...))
    else List()
  }
)

lazy val mtags = project
  .settings(
    sharedSettings,
    ...
```

Reply:
Added separateSettings to all 2.12 project.